### PR TITLE
Upload cleanup

### PR DESCRIPTION
**Pull request by sbruhns**
Allow user to delete unmentioned keys by using phraseapp-client.

@tobstarr Please review.

### CODE REVIEW

File: upload_cleanup.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	if len(keys) == 0 {
+		fmt.Println("There were no keys unmentioned in that upload.")
+		return nil
+	}
+
+	for len(keys) != 0 {
+		ids := make([]string, len(keys), len(keys))
+		names := make([]string, len(keys), len(keys))
+		for i, key := range keys {
+			ids[i] = key.ID
```
Comment (tobstarr) [inline upload_cleanup.go]: these would just be `ids = append(ids, key.ID)`

File: upload_cleanup.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		return err
+	}
+
+	if len(keys) == 0 {
+		fmt.Println("There were no keys unmentioned in that upload.")
+		return nil
+	}
+
+	for len(keys) != 0 {
+		ids := make([]string, len(keys), len(keys))
```
Comment (tobstarr) [inline upload_cleanup.go]: my (personal) taste is not not use make (in that scale the performance is not critical at all) but just `ids := []string{}` or (when using make) to use `make([]string, 0, len(keys)` and just use `append` then.

File: upload_cleanup.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		names := make([]string, len(keys), len(keys))
+		for i, key := range keys {
+			ids[i] = key.ID
+			names[i] = key.Name
+		}
+
+		if !cmd.SuppressPrompt {
+			fmt.Println("You are about to delete the following key(s) from your project:")
+			fmt.Println(strings.Join(names, " "))
```
Comment (tobstarr) [inline upload_cleanup.go]: maybe sort them `sort.Strings(names)` and print them on separate lines `Join(names, "\n")`?

File: upload_cleanup.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/phrase/phraseapp-go/phraseapp"
+)
+
+type UploadCleanupCommand struct {
+	*phraseapp.Config
+	ID             string `cli:"arg required"`
+	SuppressPrompt bool   `cli:"opt --yes desc='Don’t ask for confirmation'"`
```
Comment (tobstarr) [inline upload_cleanup.go]: I do not like `--yes` that much in this context but either go with `--confirm`, in that case we could just rename `SupressPrompt` to `Confirm` (as prompt is just the implementation of how to ask for confirmation anyway).

**Comment by tobstarr** [comment]
@sbruhns no "real" blockers (except maybe the `--yes` as are creating some kind of API contract)

**Comment by sbruhns** [comment]
@tobstarr Thanks for your feedback. Change the --yes to confirm and ordered the keys before confirm. Is there a reason why to use append instead create a correct sized array and use it?

**Comment by tobstarr** [comment]
@sbruhns no, but there is also no real performance benefit so I tend to not use make. Also: using the `make([]string, 0, len(ids)` syntax with append leads basically to the same thing (as the second parameter is the capacity so that the slice is never changed)

**Comment by alexanderwilling** [comment]
@tobstarr: true, `ids := []string{}` is more readable, and it doesn't noticeably affect performance here, so I really don't mind either way in this case.

**Pull request merged**